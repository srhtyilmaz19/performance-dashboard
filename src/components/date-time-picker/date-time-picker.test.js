import React from "react";

import {cleanup, render, waitFor, queryByAttribute} from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';
import DateTimePickers from "./date-time-picker";
import {format, subMinutes} from "date-fns";
import userEvent from "@testing-library/user-event";

afterEach(cleanup)


describe('<DateTimePickers />', () => {

    it('should render date-time-pickers', async () => {
        const nowDate = () => new Date();

        const mockDateRange = {
            start_date: subMinutes(nowDate(), 30),
            end_date: nowDate()
        };
        const {asFragment} = render(<DateTimePickers dateRange={mockDateRange} onChange={() => {
        }}/>)

        expect(asFragment()).toMatchSnapshot();
    });


    it('should contain date-time-pickers', async () => {

        const getById = queryByAttribute.bind(null, 'id');

        const nowDate = () => new Date();

        const mockDateRange = {
            start_date: subMinutes(nowDate(), 30),
            end_date: nowDate()
        };
        const dom = render(<DateTimePickers dateRange={mockDateRange} onChange={() => {
        }}/>)

        await waitFor(() => {

            const startDateTimePicker = getById(dom.container, 'start_date');
            const endDateTimePicker = getById(dom.container, 'end_date');

            expect(startDateTimePicker).toBeInTheDocument();
            expect(endDateTimePicker).toBeInTheDocument();
        });
    });


    it('should match with default values', async () => {
        const getById = queryByAttribute.bind(null, 'id');

        const nowDate = () => new Date();

        const mockDateRange = {
            start_date: subMinutes(nowDate(), 30),
            end_date: nowDate()
        };
        const dom = render(<DateTimePickers dateRange={mockDateRange} onChange={() => {
        }}/>)

        await waitFor(() => {

            const startDateTimePicker = getById(dom.container, 'start_date');
            const endDateTimePicker = getById(dom.container, 'end_date');

            expect(startDateTimePicker.value).toBe(format(mockDateRange.start_date, 'yyyy/MM/dd HH:mm'));
            expect(endDateTimePicker.value).toBe(format(mockDateRange.end_date, 'yyyy/MM/dd HH:mm'));
        });
    });


    it('should match with updated values', async () => {
        const getById = queryByAttribute.bind(null, 'id');

        const nowDate = () => new Date();

        const mockDateRange = {
            start_date: subMinutes(nowDate(), 30),
            end_date: nowDate()
        };

        const dom = render(<DateTimePickers dateRange={mockDateRange} onChange={() => {
        }}/>)

        await waitFor(() => {

            const startDateTimePicker = getById(dom.container, 'end_date');
            const endDateTimePicker = getById(dom.container, 'start_date');

            userEvent.type(startDateTimePicker, format(mockDateRange.end_date, 'yyyy/MM/dd HH:mm'))
            userEvent.type(endDateTimePicker, format(mockDateRange.start_date, 'yyyy/MM/dd HH:mm'))

            expect(startDateTimePicker).toHaveValue(format(mockDateRange.end_date, 'yyyy/MM/dd HH:mm'))
            expect(endDateTimePicker).toHaveValue(format(mockDateRange.start_date, 'yyyy/MM/dd HH:mm'))
        });

    });
});



