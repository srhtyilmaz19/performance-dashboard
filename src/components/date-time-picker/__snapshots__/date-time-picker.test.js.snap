// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateTimePickers /> should render date-time-pickers 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-container-1"
  >
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-textField-2"
    >
      <input
        class="MuiInputBase-input MuiInput-input"
        id="start_date"
        type="text"
        value="2021/05/08 15:20"
      />
    </div>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-textField-2"
    >
      <input
        class="MuiInputBase-input MuiInput-input"
        id="end_date"
        type="text"
        value="2021/05/08 15:50"
      />
    </div>
  </div>
</DocumentFragment>
`;
